jbpm-workitems/jbpm-workitems-rest/src/main/java/org/jbpm/process/workitem/rest/RESTWorkItemHandler.java

```diff
@@ -156,6 +155,7 @@
     private static final Logger logger = LoggerFactory.getLogger(RESTWorkItemHandler.class);
     private static final int DEFAULT_TOTAL_POOL_CONNECTIONS = 500;
     private static final int DEFAULT_MAX_POOL_CONNECTIONS_PER_ROUTE = 50;
+    private static final boolean DEFAULT_USE_SYSTEM_PROPERTIES = false;
```

Comment:
usually if you have a proxy what is the common thing is that everything is affecting by the environment and not really at design time. 
You need to ajust this to a system property and read from it. Also would be good if you can set/override the properties like which proxy to use
you have examples here how to do this 
https://github.com/kiegroup/droolsjbpm-integration/blob/master/kie-server-parent/kie-server-api/src/main/java/org/kie/server/api/KieServerConstants.java#L39

Reply:
The decission to use system properties or not is not base on the value of this private variable. The decision to use system properties or not is based on the value of the parameter passed to the constructor. 
This private constant is there to make easier maintance in case we decide to use system properties by default in the future. Now, for backward compatibility, the default value is false. If in future we decide to use them by default, we just need to change the constant (so this constant was introduce just becasue we have many constructor, if this class was built from scracth, we should use a builder pattern)
If we were going to control this functionality through system properties, changes in the constructor are not needed, but that was not the proposal of the JIRA, not consistent with the current structure of the code

Reply:
Rather than following the proposal of the JIRA, since the decission to use or not system properties should be taken as execution time, not desing time, it does not make sense to change current constructor, but provide a new system property org.kie.workspace.rest.useSystemProperties (default value false) which will determine if useSystemProperties is invoked or not when creating the http connection